src/domainobjectfactories/cash_flow_factory.py

```diff
@@ -0,0 +1,134 @@
+from datetime import datetime, timezone
+import random
+
+from domainobjectfactories.creatable import Creatable
+
+
+class CashFlowFactory(Creatable):
+    """ Class to create cash flows. Create method will create
+        a set amount of cash flows."""
+    PAYMENT_STATUSES = ['Actual', 'Contractual']
+
+    def create(self, record_count, start_id):
+        """ Create a set number of cash flows
+
+        Parameters
+        ----------
+        record_count : int
+            Number of cash flows to create
+        start_id : int
+            Starting id to create from
+
+        Returns
+        -------
+        List
+            Containing 'record_count' cash flows
+        """
+
+        records = []
+
+        for _ in range(start_id, start_id+record_count):
+            records.append(self.__create_record())
+
+        return records
+
+    def __create_record(self):
+        """ Create a single cash flow
+
+        Returns
+        -------
+        dict
+            A single cash flow object
+        """
+
+        record = {
+            'account_id': self.__get_account_id(),
+            'corporate_action_id': self.__create_corporate_action_id(),
+            'quantity': self.__get_quantity(),
+            'currency': self.__get_currency(),
```

Comment:
Creatable has a `create_currency` method that does the same thing. Remove the `__get_currency` method from this class and in the record use 

`'currency': self.create_currency()`